third_party/move/move-vm/runtime/src/effect_converter.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::{PartialVMError, PartialVMResult};
+use move_core_types::{value::MoveTypeLayout, vm_status::StatusCode};
+use move_vm_types::values::Value;
+
+/// Trait that defines a generic interface for converting effects of a
+/// transaction into a desired type. This allows clients of the Move VM
+/// to process effects in their own way.
+pub trait EffectConverter<R> {
+    fn convert_resource(&self, value: Value, layout: MoveTypeLayout) -> PartialVMResult<R>;
+}
```

Comment:
Do you expect to add more methods to this trait? If not, then maybe it would be cleaner to just use `Fn(Value, MoveTypeLayout) -> PartialVMResult<R>`?

Reply:
I would imagine modules, but lets use a lambda instead

Reply:
Modules are another interesting case. We tend believe it's better to preserve the original bytes users pass in and must not reserialize the module, potentially using a different version of the serializer.